pkg/artifactory/resource/replication/resource_artifactory_local_repository_single_replication.go

```diff
@@ -228,11 +224,26 @@ func resourceLocalSingleReplicationRead(_ context.Context, d *schema.ResourceDat
 		}
 		return diag.FromErr(err)
 	}
-	repConfig := GetLocalSingleReplicationBody{
-		Replication: replication,
+
+	replicationList, ok := replicationInterface.([]interface{})
+	if ok {
+		jsonData, _ := json.Marshal(replicationList)
+		var replications []getLocalSingleReplicationBody
+		json.Unmarshal(jsonData, &replications)
+
+		return packLocalSingleReplication(&replications[0], d)
 	}
 
-	return packLocalSingleReplication(&repConfig, d)
+	replicationObj, ok := replicationInterface.(interface{})
+	if ok {
+		jsonData, _ := json.Marshal(replicationObj)
+		var replication getLocalSingleReplicationBody
+		json.Unmarshal(jsonData, &replication)
+
+		return packLocalSingleReplication(&replication, d)
+	}
+
+	return nil
```

Comment:
This should return `diag.Errorf(...)` right? Because if we reach here, that means the payload can't be converted to single or array of single replication type.

Reply:
Agree.